torchtune/modules/optim/cpu_offload.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torch.optim.optimizer import ParamsT
+from torchtune.config._utils import _get_component_from_path
+
+try:
+    from torchao.prototype.low_bit_optim import (
+        CPUOffloadOptimizer as _CPUOffloadOptimizer,
+    )
+
+    _CPU_OFFLOAD_OPTIM_AVAILABLE = True
```

Comment:
For what it's worth we do have [this utility](https://github.com/pytorch/torchtune/blob/9e65fa9ae67ce09e953cb68b5093d42cb6b31310/torchtune/modules/low_precision/_utils.py#L32) to help with gating on a specific ao version. Advantage being it's a bit more explicit and that way once we upgrade to the necessary ao version it'll be easy to see that we can remove the try/except

Reply:
I'm aware of that function. But the required gating logic would be:
1. If version >= 0.5 (note that 0.4.1 will not include CPU offload optimizer, since it is a bugfix, though I don't think there will be a 0.4.1 release now)
2. OR, version == 0.4.x+gitxxxxxx i.e. install ao from source

As you can see, the check is quite burdensome, thus I want to avoid that.

`_get_torchao_version()` was mainly for handling is_fbcode and torchao-nightly, so it's not particularly helpful in this situation I think.

Reply:
Ah good point, we need to update it to account for source installs. I've been meaning to do that but haven't gotten around to it yet. No need to worry about that here then, I think what you have is fine